src/PyDynamic/uncertainty/propagate_convolution.py

```diff
@@ -167,3 +162,18 @@ def convolve_unc(x1, U1, x2, U2, mode="full"):
         raise ValueError(f'convolve_unc: Mode "{mode}" is not supported.')
 
     return conv, Uconv
+
+
+def ensure_cov_matrix(unc_array):
+    """
+    Converts 1D-arrays of standard uncertainties into the corresponding 
+    (diagonal) covariance matrix by *square*+diag. 
+
+    Does not modify inputs which are 2D-arrays or None.
+    """
+    
```

Comment:
The [static code analysis errors](https://app.codacy.com/gh/PTB-M4D/PyDynamic/pullRequest?prid=9150006) due to trailing white space should be resolved by black. At some point we should include that in the pipeline.

Reply:
Should be fixed